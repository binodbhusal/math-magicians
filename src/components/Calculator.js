import React, { useState } from 'react';
import PropTypes from 'prop-types';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [displayValue, setDisplayValue] = useState(0);
  const [calcData, setCalcData] = useState({
    total: 0,
    next: null,
    operation: null,
  });
  const handleButtonClick = (e) => {
    const buttonName = e.target.name;
    const newData = calculate(calcData, buttonName);
    setCalcData(newData);
    setDisplayValue(newData.next || newData.total || 0);
  };
  return (
    <>
      <h3>Lets Do Some Math !</h3>
      <div className="calculator">
        <Display value={String(displayValue)} />
        <div className="row-button">
          <button type="button" className="button" name="AC" onClick={handleButtonClick}>AC</button>
          <button type="button" className="button" name="+/-" onClick={handleButtonClick}>+/-</button>
          <button type="button" className="button" name="%" onClick={handleButtonClick}>%</button>
          <button type="button" className="button sign" name="&#247" onClick={handleButtonClick}>&#247;</button>
        </div>
        <div className="row-button">
          <button type="button" className="button" name="7" onClick={handleButtonClick}>7</button>
          <button type="button" className="button" name="8" onClick={handleButtonClick}>8</button>
          <button type="button" className="button" name="9" onClick={handleButtonClick}>9</button>
          <button type="button" className="button sign" name="x" onClick={handleButtonClick}>x</button>
        </div>
        <div className="row-button">
          <button type="button" className="button" name="4" onClick={handleButtonClick}>4</button>
          <button type="button" className="button" name="5" onClick={handleButtonClick}>5</button>
          <button type="button" className="button" name="6" onClick={handleButtonClick}>6</button>
          <button type="button" className="button sign" name="-" onClick={handleButtonClick}>-</button>
        </div>
        <div className="row-button">
          <button type="button" className="button" name="1" onClick={handleButtonClick}>1</button>
          <button type="button" className="button" name="2" onClick={handleButtonClick}>2</button>
          <button type="button" className="button" name="3" onClick={handleButtonClick}>3</button>
          <button type="button" className="button sign" name="+" onClick={handleButtonClick}>+</button>
        </div>
        <div className="row-button">
          <button type="button" className="button zero-button" name="0" onClick={handleButtonClick}>0</button>
          <button type="button" className="button" name="." onClick={handleButtonClick}>.</button>
          <button type="button" className="button sign" name="=" onClick={handleButtonClick}>=</button>
        </div>
      </div>
    </>
  );
};
const Display = ({ value }) => (
  <div className="display">
    <p>{value}</p>
  </div>
);

Display.propTypes = {
  value: PropTypes.string.isRequired,
};

export default Calculator;
