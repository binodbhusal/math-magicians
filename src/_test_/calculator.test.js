import React from 'react';
import { render } from '@testing-library/react';
import Calculator from '../components/Calculator';
import '@testing-library/jest-dom/extend-expect';

test('Expect homepage to match working snapshot', () => {
  const { asFragment } = render(<Calculator />);
  expect(asFragment()).toMatchSnapshot();
});
