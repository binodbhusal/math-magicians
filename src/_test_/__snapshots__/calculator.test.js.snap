// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect homepage to match working snapshot 1`] = `
<DocumentFragment>
  <h3>
    Lets Do Some Math !
  </h3>
  <div
    class="calculator"
  >
    <div
      class="display"
    >
      <p>
        0
      </p>
    </div>
    <div
      class="row-button"
    >
      <button
        class="button"
        name="AC"
        type="button"
      >
        AC
      </button>
      <button
        class="button"
        name="+/-"
        type="button"
      >
        +/-
      </button>
      <button
        class="button"
        name="%"
        type="button"
      >
        %
      </button>
      <button
        class="button sign"
        name="&#247"
        type="button"
      >
        รท
      </button>
    </div>
    <div
      class="row-button"
    >
      <button
        class="button"
        name="7"
        type="button"
      >
        7
      </button>
      <button
        class="button"
        name="8"
        type="button"
      >
        8
      </button>
      <button
        class="button"
        name="9"
        type="button"
      >
        9
      </button>
      <button
        class="button sign"
        name="x"
        type="button"
      >
        x
      </button>
    </div>
    <div
      class="row-button"
    >
      <button
        class="button"
        name="4"
        type="button"
      >
        4
      </button>
      <button
        class="button"
        name="5"
        type="button"
      >
        5
      </button>
      <button
        class="button"
        name="6"
        type="button"
      >
        6
      </button>
      <button
        class="button sign"
        name="-"
        type="button"
      >
        -
      </button>
    </div>
    <div
      class="row-button"
    >
      <button
        class="button"
        name="1"
        type="button"
      >
        1
      </button>
      <button
        class="button"
        name="2"
        type="button"
      >
        2
      </button>
      <button
        class="button"
        name="3"
        type="button"
      >
        3
      </button>
      <button
        class="button sign"
        name="+"
        type="button"
      >
        +
      </button>
    </div>
    <div
      class="row-button"
    >
      <button
        class="button zero-button"
        name="0"
        type="button"
      >
        0
      </button>
      <button
        class="button"
        name="."
        type="button"
      >
        .
      </button>
      <button
        class="button sign"
        name="="
        type="button"
      >
        =
      </button>
    </div>
  </div>
</DocumentFragment>
`;
